pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -226,24 +280,32 @@ func (r *ReconcileMemberStatus) memberOperatorHandleStatus(reqLogger logr.Logger
 		memberStatus.Status.MemberOperator = operatorStatus
 		return err
 	}
+	operatorStatus.Deployment.Name = memberDeployment.Name
+
+	// check replicas
+	if memberDeployment.Status.UnavailableReplicas > 0 {
+		errCondition := newComponentErrorCondition(reasonUnavailableReplicas, errMsgUnavailableReplicas)
+		operatorStatus.Conditions = []toolchainv1alpha1.Condition{*errCondition}
+		memberStatus.Status.MemberOperator = operatorStatus
+		return fmt.Errorf(errMsgUnavailableReplicas)
+	}
 
-	// get and check conditions of member deployment
-	conditionsReady := true
-	deploymentConditions := []appsv1.DeploymentCondition{}
+	// get and check conditions
 	for _, condition := range memberDeployment.Status.Conditions {
-		deploymentConditions = append(deploymentConditions, condition)
-		conditionsReady = conditionsReady && condition.Status == corev1.ConditionTrue
+		if condition.Status != corev1.ConditionTrue {
```

Comment:
Here is the list of current deployment condition types:
https://github.com/kubernetes/kubernetes/blob/d8fd232ea1f8c91092fb5fabb7a0f3d557a2e8fb/staging/src/k8s.io/api/apps/v1beta1/types.go#L485-L495
So, theoretically there can be `DeploymentReplicaFailure` condition with status `False`. Even if it's not what the current kube does the implementation could change in the future versions. I think it's safer to check specific condition types: `DeploymentAvailable` and `DeploymentProgressing`.

Reply:
Updated to check the specific condition types. done in 89623fdd8fbeb694d378bd2d0e1c99aa567c3d55